modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/sequenceK.kt

```diff
@@ -397,3 +399,32 @@ interface SequenceKEqK : EqK<ForSequenceK> {
   override fun <A> Kind<ForSequenceK, A>.eqK(other: Kind<ForSequenceK, A>, EQ: Eq<A>): Boolean =
     SequenceK.eq(EQ).run { this@eqK.fix().eqv(other.fix()) }
 }
+
+@extension
+interface SequenceKMonadPlus : MonadPlus<ForSequenceK>, SequenceKMonad, SequenceKAlternative {
+  override fun <A, B> Kind<ForSequenceK, A>.ap(ff: Kind<ForSequenceK, (A) -> B>): SequenceK<B> =
+    fix().ap(ff)
+
+  override fun <A, B> Kind<ForSequenceK, A>.map(f: (A) -> B): SequenceK<B> =
+    fix().map(f)
+
+  override fun <A, B, Z> Kind<ForSequenceK, A>.map2(fb: Kind<ForSequenceK, B>, f: (Tuple2<A, B>) -> Z): SequenceK<Z> =
+    fix().map2(fb, f)
+
+  override fun <A> just(a: A): SequenceK<A> =
+    SequenceK.just(a)
+}
+
+@extension
+interface SequenceKMonadLogic : MonadLogic<ForSequenceK>, SequenceKMonadPlus {
+  override fun <A> Kind<ForSequenceK, A>.splitM(): Kind<ForSequenceK, Option<Tuple2<Kind<ForSequenceK, A>, A>>> {
+    val seqA = this.fix().sequence
+    val iterA = seqA.iterator()
+
+    return if (iterA.hasNext()) {
+      just(Option.just(seqA.drop(1).k() toT iterA.next()))
+    } else {
+      just(Option.empty())
+    }
+  }
```

Comment:
> There are a few problems, especially around the sequenceK instance.

will have a look at it

> Also does WriterT have an instance for MonadLogic as well?

yes, added it. currently one test is failing. could you have a look please?
java:test://arrow.mtl.WriterTTest.MonadLogic Laws: reflect is inverse of msplit

Reply:
Turns out I was looking at an old version of MonadLogic and that had a WriterT instance which got removed because it was unlawful. Good that our tests catch that ^^ Sorry about that, we can just remove it I think.

Reply:
ok, i removed the WriterT instance

Reply:
re the sequence: i don't think it's possible to copy an iterator without knowing its implementation. so i don't think this approach is possible here. do you see any other options?
i changed it to your first suggestion. i understand it has the same performance issues, however is way nicer.